mllib/src/test/scala/org/apache/spark/ml/feature/StringIndexerSuite.scala

```diff
@@ -219,6 +219,16 @@ class StringIndexerSuite
     testDefaultReadWrite(t)
   }
 
+  test("SPARK 18698: construct IndexToString with custom uid") {
+    val uid = Identifiable.randomUID("customuid")
+    val t = new IndexToString(uid)
+      .setInputCol("myInputCol")
```

Comment:
This can be simplified.  There's no need to use randomUID, set Params, or call testDefaultReadWrite.  I'd just construct the class and check its uid.